spyder/plugins/projects/widgets/main_widget.py

```diff
@@ -867,6 +881,19 @@ def _update_explorer(self, _unused):
         """Update explorer tree"""
         self._setup_project(self.get_active_project_path())
 
+    def _get_valid_recent_projects(self):
+        """Get the list of valid recent projects."""
+        if hasattr(self, 'recent_projects'):
+            recent_projects = self.recent_projects
+        else:
+            recent_projects = self.get_conf('recent_projects', [])
```

Comment:
Could not be better for this method to have a `recent_projects` param? So it will be called to initialize `self.recent_projects` with something like:

```python
self.recent_projects = self._get_valid_recent_projects(self.get_conf('recent_projects', []))
```

And when saving the recent projects be called with something like:

```python
 self.set_conf('recent_projects', self._get_valid_recent_projects(self.recent_projects))
```